lib/composer/package/loader/project_attributes_loader.rb

```diff
@@ -0,0 +1,56 @@
+module Composer
+  module Package
+    module Loader
+      # Loads a package from the project attributes
+      class ProjectAttributesLoader < Composer::Package::Loader::ProjectLoader
+
+        def initialize(loader = nil)
+          if loader.nil?
+            loader = Composer::Package::Loader::HashLoader.new
+          end
+          @loader = loader
+        end
+
+        # Load a json string or file
+        # Param:  Project project The target gitlab project.
+        # Param:  Branch|Tag ref The target gitlab project branch/tag.
+        # Param:  Hash config A hash containing addional package configuration.
+        # Returns: Composer::Package::Package
+        def load(project, ref, type = 'library')
+          config = {
+            'name'                => project.path_with_namespace.gsub(/\s/, '').downcase,

```

Comment:
Don't know if it is specified somewhere, but each composer package on packagist is also lowercased and space free. See https://packagist.org/explore/popular. I would say it is a good idea to make it packagist compatible right from the start.